hazelcast/src/hazelcast/client/config.cpp

```diff
@@ -686,14 +690,12 @@ namespace hazelcast {
             return redo_operation_;
         }
 
-        load_balancer *const client_config::get_load_balancer() {
-            if (!load_balancer_)
-                return &default_load_balancer_;
-            return load_balancer_;
+        std::unique_ptr<load_balancer> client_config::get_load_balancer() {
```

Comment:
just in case that the user can also make this call, i think that this can only return a read-only reference and we can have another internal api for moving it to lifecycle_service or just use reference also at lifecycle_service, since the config will be kept in client instance impl until it is destroyed.

Reply:
OK, I am keeping the load balancer in the config, and holding a reference to it in ConnectionManagerImpl.